docs/docs/reference-react-component.md

```diff
@@ -155,11 +240,19 @@ The second parameter is an optional callback function that will be executed once
 
```

Comment:
>The second parameter is an optional callback function that will be executed once `setState` is completed and the component is re-rendered.

Maybe add "Generally we recommend using `componentDidUpdate()` for such logic instead."